### PR TITLE
New Resource: `azurerm_automanage_configuration`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
### Swagger Link
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/automanage/resource-manager/Microsoft.Automanage/stable/2022-05-04/automanage.json

### Specital about this resource
Property `configuration` was flatterned in to a series of properties. I am happy to take suggestions about the method handling these properties.

This first PR is only included partial json properties, more are one the way if this handling method is approved.

Context and discussion about this case: 
https://github.com/hashicorp/terraform-provider-azurerm/pull/19358

### Testing Evidence
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\Temp\GoLand\___go_test_github_com_hashicorp_terraform_provider_azurerm_internal_services_automanage.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/automanage #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\Temp\GoLand\___go_test_github_com_hashicorp_terraform_provider_azurerm_internal_services_automanage.test.exe -test.v -test.paniconexit0 #gosetup
=== RUN   TestAccAutoManageConfigurationProfile_basic
=== PAUSE TestAccAutoManageConfigurationProfile_basic
=== CONT  TestAccAutoManageConfigurationProfile_basic
--- PASS: TestAccAutoManageConfigurationProfile_basic (148.85s)
=== RUN   TestAccAutoManageConfigurationProfile_requiresImport
=== PAUSE TestAccAutoManageConfigurationProfile_requiresImport
=== CONT  TestAccAutoManageConfigurationProfile_requiresImport
--- PASS: TestAccAutoManageConfigurationProfile_requiresImport (149.39s)
=== RUN   TestAccAutoManageConfigurationProfile_complete
=== PAUSE TestAccAutoManageConfigurationProfile_complete
=== CONT  TestAccAutoManageConfigurationProfile_complete
--- PASS: TestAccAutoManageConfigurationProfile_complete (149.05s)
=== RUN   TestAccAutoManageConfigurationProfile_update
=== PAUSE TestAccAutoManageConfigurationProfile_update
=== CONT  TestAccAutoManageConfigurationProfile_update
--- PASS: TestAccAutoManageConfigurationProfile_update (189.91s)
PASS

Process finished with the exit code 0

### CODE REVIEW

**Comment by liuwuliuyun** [comment]
Will fix the lint commit soon.

File: internal/services/automanage/automanage_configuration_resource.go [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+		// "Antimalware/Exclusions/Extensions": string (extensions separated by semicolon. Ex: ".ext1;.ext2"),
+		// "Antimalware/Exclusions/Paths": string (Paths separated by semicolon. Ex: "c:\excluded-path-1;c:\excluded-path-2"),
+		// "Antimalware/Exclusions/Processes": string (Processes separated by semicolon. Ex: "proc1.exe;proc2.exe"),
+		"antimalware": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"enabled": {
```
Comment (stephybun) [inline internal/services/automanage/automanage_configuration_resource.go:95]: We shouldn't be exposing this field for blocks - the behaviour for features turned on/off should be if block is present -> feature is on, block is absent -> feature is off. We have a [section on this in the contributor docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#features-that-are-toggled-by-the-property-enabled).

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/automanage/automanage_configuration_resource.go (hunk 1 above)
Comment (liuwuliuyun) [reply internal/services/automanage/automanage_configuration_resource.go]: Thanks for the review, update according to the guide. Removing the enable property.

**Comment by liuwuliuyun** [comment]
=== CONT  TestAccAutoManageConfigurationProfile_requiresImport
--- PASS: TestAccAutoManageConfigurationProfile_requiresImport (328.52s)
=== RUN   TestAccAutoManageConfigurationProfile_complete
=== PAUSE TestAccAutoManageConfigurationProfile_complete
=== CONT  TestAccAutoManageConfigurationProfile_complete
--- PASS: TestAccAutoManageConfigurationProfile_complete (365.85s)
=== RUN   TestAccAutoManageConfigurationProfile_update
=== PAUSE TestAccAutoManageConfigurationProfile_update
=== CONT  TestAccAutoManageConfigurationProfile_update
--- PASS: TestAccAutoManageConfigurationProfile_update (459.94s)
PASS


Process finished with the exit code 0

File: internal/services/automanage/automanage_configuration_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+func (r AutoManageConfigurationProfileResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+				%s
+
+resource "azurerm_automanage_configuration" "test" {
+  name                = "acctest-amcp-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%s"
+  antimalware {
```
Comment (stephybun) [inline internal/services/automanage/automanage_configuration_resource_test.go:154]: Basic test config should only contains properties that are Required but this block is marked as Optional so this should be removed?

**Review by stephybun — CHANGES_REQUESTED** [review]
In addition to the comment on the basic test config - can we add a test that toggles the antimalware feature on/off through the addition and removal of the block to verify that works?

File: internal/services/automanage/automanage_configuration_resource_test.go (hunk 2 above)
Comment (liuwuliuyun) [reply internal/services/automanage/automanage_configuration_resource_test.go:190]: Agreed. Added test case antimalware and remove it configuration in basic.

**Comment by liuwuliuyun** [comment]
=== PAUSE TestAccAutoManageConfigurationProfile_basic
=== CONT  TestAccAutoManageConfigurationProfile_basic
--- PASS: TestAccAutoManageConfigurationProfile_basic (150.38s)
=== RUN   TestAccAutoManageConfigurationProfile_antimalware
=== PAUSE TestAccAutoManageConfigurationProfile_antimalware
=== CONT  TestAccAutoManageConfigurationProfile_antimalware
--- PASS: TestAccAutoManageConfigurationProfile_antimalware (190.12s)
=== RUN   TestAccAutoManageConfigurationProfile_requiresImport
=== PAUSE TestAccAutoManageConfigurationProfile_requiresImport
=== CONT  TestAccAutoManageConfigurationProfile_requiresImport
--- PASS: TestAccAutoManageConfigurationProfile_requiresImport (152.70s)
=== RUN   TestAccAutoManageConfigurationProfile_complete
=== PAUSE TestAccAutoManageConfigurationProfile_complete
=== CONT  TestAccAutoManageConfigurationProfile_complete
--- PASS: TestAccAutoManageConfigurationProfile_complete (94.15s)
=== RUN   TestAccAutoManageConfigurationProfile_update
=== PAUSE TestAccAutoManageConfigurationProfile_update
=== CONT  TestAccAutoManageConfigurationProfile_update
--- PASS: TestAccAutoManageConfigurationProfile_update (201.93s)
PASS


Process finished with the exit code 0

**Comment by stephybun** [comment]
@liuwuliuyun - [last time this PR was opened](https://github.com/hashicorp/terraform-provider-azurerm/pull/19358) there were discussions around the [`ConfigurationDictionary`](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/automanage/resource-manager/Microsoft.Automanage/stable/2022-05-04/automanage.json#LL2147C5-L2147C5) defined in the swagger. 

My expectation with the reopening of this is that the properties exposed in the resource's schema are now defined in the swagger as they should be according to [this comment thread](https://github.com/hashicorp/terraform-provider-azurerm/pull/19358#discussion_r1037177546). We should go back to the service team and have that fixed. 

Since we can't accept this in it's current state and without the properties being explicitly defined in the swagger I'm going to close this PR.

**Comment by liuwuliuyun** [comment]
@stephybun I thought we had another round of discussion and agreed to proceed with with current status with service team supply the schema without changing the swagger. Let me recheck with our PM.

**Review by stephybun — APPROVED** [review]
Re-opening after internal discussion. Thanks @liuwuliuyun LGTM 🏔️

**Pull request merged**